modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/RefNameDeclarationEnricher.java

```diff
@@ -60,33 +51,21 @@ public void onConfiguration(ConfigurationDeclaration declaration) {
       protected void onConnectionProvider(ConnectionProviderDeclaration declaration) {
         doEnrich(declaration);
       }
+
     }.walk(extensionLoadingContext.getExtensionDeclarer().getDeclaration());
   }
 
-  private void doEnrich(BaseDeclaration<?> declaration) {
-
+  @Override
+  protected void doEnrich(BaseDeclaration<?> declaration) {
     declaration.getModelProperty(ImplementingTypeModelProperty.class).ifPresent(typeProperty -> {
       Collection<Field> fields = getAllFields(typeProperty.getType(), withAnnotation(RefName.class));
```

Comment:
there's still a lot of duplicated code. Can't the part of fetching the ImplementType and extracting the fields also be reused by simply passing the Annotation class to filter upon?